src/Disks/ObjectStorages/DiskObjectStorage.cpp

```diff
@@ -725,6 +725,7 @@ std::unique_ptr<ReadBufferFromFileBase> DiskObjectStorage::readFile(
 
     /// Avoid cache fragmentation by choosing bigger buffer size.
     bool prefer_bigger_buffer_size = read_settings.filesystem_cache_prefer_bigger_buffer_size
+        && !read_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache
```

Comment:
I guess this should help with memory consumption for merges, can you please explain how?

Reply:
btw @al13n321 should we have something similar for userspace page cache, e.g. read-only mode of page cache for merges?